tests/devices/test_null_qubit.py

```diff
@@ -11,1377 +11,1248 @@
+        assert isinstance(result, tuple)
 
-    def test_paulix_pauliy(self, nullqubit_device, theta, phi, varphi):
-        """Test that a tensor product involving PauliX and PauliY works correctly"""
-        dev = nullqubit_device(wires=3)
+        exp, probs, sample = result
+        assert np.array_equal(exp, 0.0)
+        assert np.array_equal(probs, [1.0, 0.0, 0.0, 0.0])
+        assert np.array_equal(sample, np.zeros((10000, 2)))
 
-        obs = qml.PauliX(0) @ qml.PauliY(2)
+    shots_data = [
+        [10000, 10000],
+        [(10000, 2)],
+        [10000, 20000],
+        [(10000, 2), 20000],
+        [(10000, 3), 20000, (30000, 2)],
+    ]
 
-        dev.apply(
-            [
-                qml.RX(theta, wires=[0]),
-                qml.RX(phi, wires=[1]),
-                qml.RX(varphi, wires=[2]),
-                qml.CNOT(wires=[0, 1]),
-                qml.CNOT(wires=[1, 2]),
-            ],
-            obs.diagonalizing_gates(),
+    @pytest.mark.parametrize("shots", shots_data)
+    def test_expval_shot_vector(self, shots):
+        """Test a simple circuit with a single expval measurement for shot vectors"""
+        x = np.array(0.732)
+        shots = qml.measurements.Shots(shots)
+        qs = qml.tape.QuantumScript([qml.RY(x, wires=0)], [qml.expval(qml.PauliZ(0))], shots=shots)
+
+        dev = NullQubit()
+        result = dev.execute(qs)
+
+        assert isinstance(result, tuple)
+        assert len(result) == len(list(shots))
+        assert all(r == 0.0 for r in result)
+
+    @pytest.mark.parametrize("shots", shots_data)
+    def test_probs_shot_vector(self, shots):
+        """Test a simple circuit with a single prob measurement for shot vectors"""
+        x = np.array(0.732)
+        shots = qml.measurements.Shots(shots)
+        qs = qml.tape.QuantumScript([qml.RY(x, wires=0)], [qml.probs(wires=0)], shots=shots)
+
+        dev = NullQubit()
+        result = dev.execute(qs)
+
+        assert isinstance(result, tuple)
+        assert len(result) == len(list(shots))
+
+        assert all(np.array_equal(res, [1.0, 0.0]) for res in result)
+
+    @pytest.mark.parametrize("shots", shots_data)
+    def test_sample_shot_vector(self, shots):
+        """Test a simple circuit with a single sample measurement for shot vectors"""
+        x = np.array(0.732)
+        shots = qml.measurements.Shots(shots)
+        qs = qml.tape.QuantumScript([qml.RY(x, wires=0)], [qml.sample(wires=range(2))], shots=shots)
+
+        dev = NullQubit()
+        result = dev.execute(qs)
+
+        assert isinstance(result, tuple)
+        assert len(result) == len(list(shots))
+        assert all(np.array_equal(res, np.zeros((s, 2))) for res, s in zip(result, shots))
+
+    @pytest.mark.parametrize("shots", shots_data)
+    def test_multi_measurement_shot_vector(self, shots):
+        """Test a simple circuit containing multiple measurements for shot vectors"""
+        # TODO: resume from here
```

Comment:
Resume what? (For context, I'm working on cleaning up out-dated TODOs in the test suite)

Reply:
@astralcai I can say with ~75% confidence that this can be removed

I often write myself a TODO like that at the end of the day if I'm halfway through a large tedious task (eg. refactor, or in this case a copy-paste-rewrite of an existing test file). I probably just forgot to delete it the next day and it made it onto master. sorry about that!